### ISSUE TITLE
ERROR o.apache.tomcat.util.net.NioEndpoint - java.lang.OutOfMemoryError: Java heap space

### ISSUE THREAD

**Issue by chaitanmk** [hazelcast/hazelcast#10717]
hi,

we are frequently getting OutOfMemoryError in our application. We have configured below eviction rules in hazlecast. but those are not working. Can you please advice best practices to overcome heap space isssue.
```xml
                <in-memory-format>BINARY</in-memory-format>
		<async-backup-count>0</async-backup-count>
		<time-to-live-seconds>0</time-to-live-seconds>
		<max-idle-seconds>1800</max-idle-seconds>
		<eviction-policy>LFU</eviction-policy>
		<max-size policy="USED_HEAP_PERCENTAGE">25</max-size>
		<eviction-percentage>25</eviction-percentage>
		<min-eviction-check-millis>100</min-eviction-check-millis>
		<merge-policy>com.hazelcast.map.merge.PassThroughMergePolicy</merge-policy>
```
thank in advance.

**Comment by ahmetmircik** [comment]
Hi, what was the hazelcast version?
Also it would be good to test against latest released version 3.8.2 to see any possible fix.

**Comment by ahmetmircik** [comment]
Also `USED_HEAP_PERCENTAGE` is not an appropriate policy if you configure multiple maps with it. It can cause OOME. You can consider `FREE_HEAP_PERCENTAGE` instead.

Closing this now, if you still see the problem after the suggestions here please reopen this issue.
